### PR TITLE
feat: feed gallery attachments

### PR DESCRIPTION
**Pull request by ice-brontes**
## Description
Implemented photo gallery attachments in feed.

## Additional Notes
1. Added carousel media preview for attached photo/video in reply and post during creation.
2. Increased max size for the reply input field.
3. Changed the attached media in the article. Now it takes all the width of the text field.
4. Moved `calculatePostImageAspectRatio` from `Post` to the general utils for reusing.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Refactoring
- [ ] Documentation
- [ ] Chore

## Video Presentation
https://github.com/user-attachments/assets/6ac4f02d-56ce-494b-a518-82ba7ab32adb

### CODE REVIEW

File: lib/app/features/feed/create_post/views/pages/create_post_modal/components/attached_media_preview_list.dart [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  });
+
+  final ValueNotifier<List<MediaFile>> attachedMediaNotifier;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final list = attachedMediaNotifier.value;
+
+    final maxItemHeight = 190.0.s;
+    final attachedMediaAspectRatio = list.isEmpty
```
Comment (ice-alcides) [inline lib/app/features/feed/create_post/views/pages/create_post_modal/components/attached_media_preview_list.dart:26]: I suppose the lines 26-31 can be moved outside of UI

Comment (ice-brontes) [reply lib/app/features/feed/create_post/views/pages/create_post_modal/components/attached_media_preview_list.dart:26]: To be honest, I'm not sure. `attachedMediaAspectRatio`, `isHorizontalPreviews` directly related to the current widget's UI and will not be reused anywhere else. Can you suggest the better place for it?

Comment (ice-brontes) [reply lib/app/features/feed/create_post/views/pages/create_post_modal/components/attached_media_preview_list.dart]: Added provider for that and class wrapper to handle `isHorizontal`, `isVertical` stuff.

**Review by ice-alcides — DISMISSED** [review]

**Review by ice-kreios — APPROVED** [review]

File: lib/app/features/feed/create_post/views/pages/create_post_modal/components/attached_media_preview_list.dart [hunk 2]
```diff
@@ -23,12 +23,11 @@ class AttachedMediaPreview extends ConsumerWidget {
     final list = attachedMediaNotifier.value;
 
     final maxItemHeight = 190.0.s;
-    final attachedMediaAspectRatio = list.isEmpty
-        ? 0.0
-        : calculateMediaAspectRatio(
-            ratioProviders: list.map(MediaAspectRatio.fromMediaFile),
-          );
-    final isHorizontalPreviews = attachedMediaAspectRatio >= 1;
+    final attachedMediaAspectRatio = ref.read(
```
Comment (ice-alcides) [inline lib/app/features/feed/create_post/views/pages/create_post_modal/components/attached_media_preview_list.dart:26]: ref.watch

**Review by ice-kreios — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**